example/spec/ingestGranule/IngestUMMGSuccessSpec.js

```diff
@@ -209,65 +239,114 @@ describe('The S3 Ingest Granules workflow configured to ingest UMM-G', () => {
   });
 
   describe('the PostToCmr task', () => {
-    let cmrResource;
-    let cmrLink;
-    let response;
+    let bucketsConfig;
+    let onlineResources;
     let files;
-    let resourceHrefs;
+    let resourceURLs;
 
     beforeAll(async () => {
+      bucketsConfig = new BucketsConfig(config.buckets);
+
       postToCmrOutput = await lambdaStep.getStepOutput(workflowExecution.executionArn, 'PostToCmr');
       if (postToCmrOutput === null) throw new Error(`Failed to get the PostToCmr step's output for ${workflowExecution.executionArn}`);
 
-      files = postToCmrOutput.payload.granules[0].files;
-      cmrLink = postToCmrOutput.payload.granules[0].cmrLink;
-      cmrResource = await getOnlineResources(cmrLink);
-      response = await got(cmrResource[2].href);
+      granule = postToCmrOutput.payload.granules[0];
+      files = granule.files;
 
-      resourceHrefs = cmrResource.map((resource) => resource.href);
+      onlineResources = await getOnlineResources(granule);
+      resourceURLs = onlineResources.map((resource) => resource.URL);
     });
 
     it('has expected payload', () => {
-      granule = postToCmrOutput.payload.granules[0];
       expect(granule.published).toBe(true);
-      expect(granule.cmrLink.startsWith('https://cmr.uat.earthdata.nasa.gov/search/granules.json?concept_id=')).toBe(true);
+      expect(granule.cmrLink).toEqual(`${getUrl('search')}granules.json?concept_id=${granule.cmrConceptId}`);
 
-      // Set the expected cmrLink to the actual cmrLink, since it's going to
-      // be different every time this is run.
-      const updatedExpectedpayload = cloneDeep(expectedPayload);
-      updatedExpectedpayload.granules[0].cmrLink = postToCmrOutput.payload.granules[0].cmrLink;
+      // Set the expected CMR values since they're going to be different
+      // every time this is run.
+      const updatedExpectedPayload = cloneDeep(expectedPayload);
+      updatedExpectedPayload.granules[0].cmrLink = granule.cmrLink;
+      updatedExpectedPayload.granules[0].cmrConceptId = granule.cmrConceptId;
 
-      expect(postToCmrOutput.payload).toEqual(updatedExpectedpayload);
+      expect(postToCmrOutput.payload).toEqual(updatedExpectedPayload);
     });
 
     it('publishes the granule metadata to CMR', async () => {
-      const result = await conceptExists(granule.cmrLink);
+      const result = await conceptExists(granule.cmrLink, true);
 
       expect(granule.published).toEqual(true);
       expect(result).not.toEqual(false);
     });
 
     it('updates the CMR metadata online resources with the final metadata location', () => {
-      const distEndpoint = config.DISTRIBUTION_ENDPOINT;
-      const extension1 = urljoin(files[0].bucket, files[0].filepath);
-      const filename = `https://${files[2].bucket}.s3.amazonaws.com/${files[2].filepath}`;
+      // Is this test still necessary?
```

Comment:
why wouldn't it be anymore?  I could be wrong, I thought this was testing that the metadata we post to CMR has the correct locations that the files were moved to in the move-granules step. ?

Reply:
I thought this test was just verifying that URLs exist in UMM-G for public files (s3 https links) and protected files (via the distribution API). Since `post-to-cmr` comes after `move-granules`, it is effectively testing that CMR was properly updated with the final locations of the files.

My initial thought was that the new test I've added to actually downloads the science URLs, it is implicitly verifying the validity of those URLs in the process. Upon further reflection, it is really just verifying that the URLs can be used to download the exact same data that was ingested, but not necessarily that those URLs match what we're expecting. So yes, I think this test is still necessary.